src/with_ladj.jl

```diff
@@ -84,9 +84,19 @@ _with_ladj_on_mapped(map_or_bc::Function, y_with_ladj::Tuple{Any,Real}) = y_with
 function _with_ladj_on_mapped(map_or_bc::Function, y_with_ladj)
```

Comment:
Julia does not specialize on `Function` arguments, so possibly this improves type inference and performance:
```suggestion
function _with_ladj_on_mapped(map_or_bc::F, y_with_ladj) where {F<:Union{typeof(map),typeof(broadcast)}}
```

Reply:
One could also use `F<:Function` but I thought it's nice to indicate once more that we only use `map` and `broadcast`.